core/services/ardupilot_manager/mavlink_proxy/MAVLinkRouter.py

```diff
@@ -24,24 +24,25 @@ def _get_version(self) -> Optional[str]:
     def assemble_command(self, master: Endpoint) -> str:
         # Convert endpoint format to mavlink-router format
         def convert_endpoint(endpoint: Endpoint) -> str:
-            if endpoint.connection_type != EndpointType.TCP:
-                return str(endpoint)[str(endpoint).find(":") + 1 :]
             # TCP uses a special argument and only works with localhost
-            port = str(endpoint).split(":")[2]
-            return f"-t {port}"
+            if endpoint.connection_type == EndpointType.TCPServer:
+                return f"--tcp-port {endpoint.argument}"
+            if endpoint.connection_type == EndpointType.TCPClient:
+                return f"--tcp-endpoint {endpoint.place}:{endpoint.argument}"
+            if endpoint.connection_type == EndpointType.UDPServer:
+                return f"--endpoint {endpoint.place}:{endpoint.argument}"
+            raise NotImplementedError(f"Endpoint of type {endpoint.connection_type} not supported on MavlinkRouter.")
 
-        endpoints = " ".join(["--endpoint " + convert_endpoint(endpoint) for endpoint in self.endpoints()])
+        endpoints = " ".join([convert_endpoint(endpoint) for endpoint in self.endpoints()])
 
         if master.connection_type not in [
             EndpointType.UDPServer,
-            EndpointType.UDPClient,
             EndpointType.Serial,
```

Comment:
for anything other than TCPServer we need to set tcp-port to 0 to avoid the bind behaviour